### PR TITLE
reorder and improve clone test to match docs

### PR DESCRIPTION
**Pull request by rmynar**
Added coverage for SAT-27907.

Reordered steps to match documentation - Pulp content has to be copied to target machine before the satellite-clone is executed.

Added steps that occupy source satellite userids on target satellite server to check that satellite-clone changes ownership and permissions.

### CODE REVIEW

File: tests/foreman/destructive/test_clone.py [hunk 1]
```diff
@@ -83,6 +93,17 @@ def test_positive_clone_backup(
     assert target_sat.cli.Service.stop().status == 0
     assert target_sat.cli.Service.disable().status == 0
 
+    # If --skip-pulp-data make sure you can rsync /var/lib/pulp over per BZ#2013776
+    if skip_pulp:
+        # Copying satellite pulp data to target RHEL
+        assert (
+            target_sat.execute(
+                f'sshpass -p "{SSH_PASS}" rsync -e "ssh -o StrictHostKeyChecking=no" --archive --partial --progress --compress '
+                f'/var/lib/pulp root@{sat_ready_rhel.hostname}:/var/lib/'
```
Comment (evgeni) [inline tests/foreman/destructive/test_clone.py:102]: The [docs](https://docs.redhat.com/en/documentation/red_hat_satellite/6.15/html/administering_red_hat_satellite/cloning_satellite_server#sec-Pulp_Data_Considerations) say:
```
# rsync --archive --partial --progress --compress \
/var/lib/pulp/ target_server.example.com:/var/lib/pulp/
```
So this should be:
```suggestion
                f'sshpass -p "{SSH_PASS}" rsync -e "ssh -o StrictHostKeyChecking=no" --archive --partial --progress --compress '
                f'/var/lib/pulp/ root@{sat_ready_rhel.hostname}:/var/lib/'
```

**Review by evgeni — APPROVED** [review]
the re-ordering is correct, just a nitpick on the rsync command to be exactly as in docs (rsync is picky about trailing slashes)

**Comment by rmynar** [comment]
trigger: test-robottelo
pytest: tests/foreman/destructive/test_clone.py::test_positive_clone_backup

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 8983
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/destructive/test_clone.py::test_positive_clone_backup --external-logging
Test Result : ================ 4 passed, 578 warnings in 10533.34s (2:55:33) =================
```

**Review by evgeni — APPROVED** [review]

**Review by LadislavVasina1 — APPROVED** [review]

**Comment by LadislavVasina1** [comment]
@rmynar don't you want to add `AutoMerge_Cherry_Picked` label to this PR?

**Review by pondrejk — APPROVED** [review]

**Pull request merged**